### ISSUE TITLE
Installation & parameter file

### ISSUE THREAD

**Issue by ingridvanwelie** [spyking-circus/spyking-circus#40]
Hi, 

I am a new user and after installing Miniconda and Spyking Circus, I am trying to run it on a datafile. I didn't have a parameter file yet, so I told the script to make one, but it is failing to do so. 

Here is the log:

Do you want SpyKING CIRCUS to create a parameter file? [Y/n] Y
Creating /Users/lawelie/lawelie/Desktop/data/botheyes/170816_145415.params
Fill it properly before launching the code! (see documentation)
-------------------------  Informations  -------------------------
| Keep in mind that filtering is performed on site, so please
| be sure to keep a copy of your data elsewhere
------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lawelie/miniconda3/bin/spyking-circus", line 11, in <module>
    load_entry_point('spyking-circus==0.6.0', 'console_scripts', 'spyking-circus')()
  File "/Users/lawelie/miniconda3/lib/python3.6/site-packages/circus/scripts/launch.py", line 147, in main
    shutil.copyfile(config_file, file_params)
  File "/Users/lawelie/miniconda3/lib/python3.6/shutil.py", line 121, in copyfile
    with open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/lawelie/lawelie/Desktop/data/botheyes/170816_145415.params'

Any suggestion as to how I can proceed?
Even if I put a parameter file in the folder myself (that I found on this repository), it doesn't run. 

Best regards,
Ingrid

**Comment by yger** [comment]
Strange... You are using Python3.6 on a Mac? Can you open File
"/Users/lawelie/miniconda3/lib/python3.6/site-packages/circus/scripts/launch.py",
line 147, and add a line, before (with the same tabulation as L147), with

print(config_file, file_params)

Tell me if the printed path are valid paths, where the files are. I'll have
a look on my side otherwise, trying to install Python3.6

Best

Pierre



2017-09-11 19:58 GMT+02:00 ingridvanwelie <notifications@github.com>:

**Comment by ingridvanwelie** [comment]
Dear Pierre,

Thank you for the quick response! 

Yes, I am using a new Powerbook with Sierra 10.12.5 and when installing Miniconda, I’d figured I would just install the most recent Python version (am new to using Python as well). I downloaded it here:

https://conda.io/miniconda.html <https://conda.io/miniconda.html>

I added the line in the script and when I ran it again, it didn’t complain about the parameter file, but now I get the following error message:

The probe file /home/pierre/spyking-circus/probes/dan.prb can not be found

So I didn’t get the path for the config file, but it is in the spyking circus folder under user (together with the probes folder). 

Hope this makes any sense. 

Best regards,
Ingrid

**Comment by yger** [comment]
So the first error is gone? Good.

However, it looks like you did not configure the values in your .params
file: you shoud not have something with /home/pierre/....  You should put
the path of your own probe file in the mapping field. Have a look to the
documentation , but you should specify the dtype, the probe file, and so on
in the created .params file

Best

Pierre



2017-09-11 22:10 GMT+02:00 ingridvanwelie <notifications@github.com>:

**Comment by ingridvanwelie** [comment]
Many thanks for the feedback!

I changed the parameters in the .params file, and put the probe file in the same folder as the data, and now it doesn’t give me an error message, but simply states: 

Permission denied. 

Ingrids-MBP:~ lawelie$ /Users/lawelie/spyking-circus/probes/mea_252.prb 
-bash: /Users/lawelie/spyking-circus/probes/mea_252.prb: Permission denied

**Comment by yger** [comment]
I'm not sure to understand, and I think that you are not properly using the
software. You should use the Spyking-circus command, not just try to launch
a .prb file. Please have a look to the quickstart described here
http://spyking-circus.readthedocs.io/en/latest/code/quickstart.html

Otherwise, the permission error may be related to read/write permissions in
your folder

**Comment by ingridvanwelie** [comment]
Dear Pierre,

The previous issue I emailed you about has been resolved, and I am now able to run the algorithm on the data. 

I am using Intan files, but got errors about the wrapper when using the rhd format, so am using the raw_binary format instead and specifying the correct parameters. See attached for my config and parameter files. Also attached my probe file - we are using custom new probes developed here at MIT.  

I am able to filter and preview the data seemingly correctly, but when clustering, the algorithm seems to get stuck on the error code below. It keeps on repeating it. Can you explain to me what it means? 

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 6: ordinal not in range(128)

For reference, I am also attaching the log file. 

Best regards,
Ingrid

**Comment by yger** [comment]
Dear Ingrid

This bug should have been fixed in the master branch (so in 0.6.1 that I'll push soon with conda/pip). Could you please update your software using the command

>> pip install git+https://github.com/spyking-circus/spyking-circus

You must have git installed, but once this is done, check that the version of the software (in the header when launched) is 0.6.1. Let me know if this solves the problem. 

Regarding the error with the intan wrapper and RHD, can you send it to me?

Best

Pierre

**Comment by ingridvanwelie** [comment]
Hi Pierre,

I did the update and it stated in the terminal that 0.6.1 is successfully installed. However, when I launch the GUI, it keeps giving me the error below in the terminal despite launching the GUI:

dhcp-18-189-35-122:~ lawelie$ spyking-circus-launcher
Sep 14 10:06:17  python[14291] <Error>: clip: empty path.

 I don’t see the version mentioned anywhere in the header while launching. 

The error I get when using the rhd format is:


---------------------------- Error -----------------------------
| There is an error in the _read_from_header method of the wrapper
'utf-8' codec can't decode byte 0x91 in position 2: invalid start byte
------------------------------------------------------------------
---------------------------- Error -----------------------------
| Shape of the data is not defined. Are you sure of the wrapper?
------------------------------------------------------------------

Process exited with exit code 1

Best,
Ingrid

**Comment by yger** [comment]
Ok, so we are slowly getting there. You are now telling me that the software is running, despite the fact that you have a weird error message about clip, am I right?. If this is the case, then this a good step, and you should already be able to analyze data, while I'm understanding what is this error (I don't think it has anything to do with the software). 
I'll also have a look to the RHD problem. You are using Mac OS, isn't it ?

**Comment by ingridvanwelie** [comment]
Ok great, so I will run my data file again and see what happens next! 

And yes, I am using Mac OS Sierra 10.12.5. 

On the acquisition end, I am using a 1024 channel Intan amp and see the link below for more details on their file formats just in case you weren’t aware of this info yet:

http://intantech.com/files/Intan_RHD2000_data_file_formats.pdf <http://intantech.com/files/Intan_RHD2000_data_file_formats.pdf>

**Comment by yger** [comment]
The clip error has nothing to do with the software, and it should not interfer. Regarding the .rhd problem, could you send me your data file if not too big? It may be easier to understand where is the problem. SpyKING CIRCUS is using the INTAN loader in the background, but it looks like there is some encoding issues

**Comment by ingridvanwelie** [comment]
Sure, I can share a dropbox folder with you with some data in it. The are pretty big files (~900MB). 

What email can I use for you to share a folder?

**Comment by ingridvanwelie** [comment]
I am running 0.6.1, but still getting this error that should have been fixed:

Traceback (most recent call last):
  File "/Users/lawelie/miniconda3/lib/python3.6/site-packages/circus/scripts/launch_gui.py", line 556, in append_error
    lines = strip_ansi_codes(to_str(self.process.readAllStandardError()))
  File "/Users/lawelie/miniconda3/lib/python3.6/site-packages/circus/scripts/launch_gui.py", line 57, in to_str
    return str(b, encoding='ascii')

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 6: ordinal not in range(128)

**Comment by yger** [comment]
Ok, let me test the graphical launcher with python3.6 (I only tested the command line)

**Comment by ingridvanwelie** [comment]
Hi Pierre,

I have another question for you. 

Everytime I run a data set in the preview mode in the GUI, I see a result like the one in the attached screenshot. 

There are some strange peaks that appear periodically at about 7 Hz that look like artifacts. There is also a huge offset artifact in the beginning of the recording that may be the cause of the periodic peaks. 

These peaks are not present when I look at the same raw data or at high-pass filtered data manually in Matlab. The initial offset at t = 0 is there in the raw data however.

The periodic peaks are unaffected by changing the filtering settings, so my suspicion is that they are caused either in the whitening step, or in the filtering step as a result of this initial offset artifact. 

Have you seen anything like this before and do you have any suggestion how to deal with it? Is there e.g. an option to exclude the first few ms of data? 

Best wishes,
Ingrid

**Comment by yger** [comment]
Dear Ingrid.

I can not find the attached image. Can you send it to me, and/or also send
me the raw data (for fixing the rhd bug)? Should I also understand that the
previous fix I told you worked?

Best

Pierre

**Comment by ingridvanwelie** [comment]
Hi Pierre,

Ok, here is another attempt at attaching the screenshot!

For sending you data: the files are pretty big (~900 MB), so I can share them with you via a dropbox folder, but which email address should I use for this? Your email address above may not be the best for sharing data?

The previous bug is still there when using the GUI, but not when I use command line. 

Best,
Ingrid

**Comment by mstimberg** [comment]
Hi Ingrid,

> Ok, here is another attempt at attaching the screenshot!

you cannot send attachments to github issues via mail, you'll have to go the website (https://github.com/spyking-circus/spyking-circus/issues/40) and add it there: https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/ (or send it via mail directly to Pierre of course ;-) )

**Comment by ingridvanwelie** [comment]
Yes thanks, I asked for Pierre's direct email twice, but he hasnt given it yet.

Can you tell me what it is?

Sent from my iPhone

On Sep 15, 2017, at 7:56 AM, Marcel Stimberg <notifications@github.com<mailto:notifications@github.com>> wrote:


Hi Ingrid,

Ok, here is another attempt at attaching the screenshot!

you cannot send attachments to github issues via mail, you'll have to go the website (#40<https://github.com/spyking-circus/spyking-circus/issues/40>) and add it there: https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/ (or send it via mail directly to Pierre of course ;-) )

**Comment by yger** [comment]
You can send me everything to pierre.yger@gmail.com

**Comment by surygeng** [comment]
> I'm not sure to understand, and I think that you are not properly using the software. You should use the Spyking-circus command, not just try to launch a .prb file. Please have a look to the quickstart described here http://spyking-circus.readthedocs.io/en/latest/code/quickstart.html Otherwise, the permission error may be related to read/write permissions in your folder

Hi Pierre, 

I see you talked about read/write permissions here. I run into a similar problem when I analysis the generated files. I use your example scripts from https://spyking-circus.readthedocs.io/en/latest/advanced/analysis.html in PyCharm. To use the spyking-circus package in Pycharm, I added the external library pointing to the anaconda. I have the generated files put in a folder that has both read and write permission. Then I assigned the absolute path to 'dat_file'. But when I run the script, it gives me an error saying "Not all nodes can read/write the data file. Check path?"

I also tried to move the generated files to the same anaconda3 folder as where the external library is, I still get the error. Where should I save and use these example scripts?
